server/endpoints/AdminActions.ts

```diff
@@ -0,0 +1,189 @@
+import { body } from "express-validator";
+
+import { getCrossListOR, getMetricValues } from "common/CourseCard";
+import { Endpoint } from "../endpoints";
+import { Reviews, ReviewDocument, Classes } from "../dbDefs";
+import { updateProfessors, findAllSemesters, resetProfessorArray } from "../dbInit";
+import { getCourseById, verifyToken } from "./utils";
+
+// The type for a request with an admin action for a review
+interface AdminReviewRequest {
+    review: ReviewDocument;
+    token: string;
+}
+
+// The type for a request with an admin action for updating professors info
+interface AdminProfessorsRequest {
+    token: string;
+}
+
+// This updates the metrics for an individual class given its Mongo-generated id.
+// Returns 1 if successful, 0 otherwise.
+export const updateCourseMetrics = async (courseId, token) => {
+  try {
+    const userIsAdmin = await verifyToken(token);
+    if (userIsAdmin) {
+      const course = await getCourseById({ courseId });
+      if (course) {
+        const crossListOR = getCrossListOR(course);
+        const reviews = await Reviews.find({ visible: 1, reported: 0, $or: crossListOR }, {}, { sort: { date: -1 }, limit: 700 }).exec();
+        const state = getMetricValues(reviews);
+        await Classes.updateOne({ _id: courseId },
+          {
+            $set: {
+            // If no data is available, getMetricValues returns "-" for metric
+              classDifficulty: (state.diff !== "-" && !isNaN(Number(state.diff)) ? Number(state.diff) : null),
+              classRating: (state.rating !== "-" && !isNaN(Number(state.rating)) ? Number(state.rating) : null),
+              classWorkload: (state.workload !== "-" && !isNaN(Number(state.workload)) ? Number(state.workload) : null),
+            },
+          });
+        return 1;
+      }
+      return 0;
+    }
+    return 0;
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log("Error: at 'updateCourseMetrics' method");
+    // eslint-disable-next-line no-console
+    console.log(error);
+    return 0;
+  }
+};
+
+/*
+ * Approve a submitted review and make it visible
+ */
+export const makeReviewVisible: Endpoint<AdminReviewRequest> = {
+  guard: [body("review").notEmpty(), body("token").notEmpty().isAscii()],
+  callback: async (adminReviewRequest: AdminReviewRequest) => {
+    try {
+      // check: make sure review id is valid and non-malicious
+      const userIsAdmin = await verifyToken(adminReviewRequest.token);
+      const regex = new RegExp(/^(?=.*[A-Z0-9])/i);
+      if (regex.test(adminReviewRequest.review._id) && userIsAdmin) {
```

Comment:
Something @khemritolya demonstrated is that we can replace these regex checks by doing that check in the guard: [example from #214 ](https://github.com/cornell-dti/course-reviews-react-2.0/pull/214/files#diff-1d08deae3ced372c187d247f952b042240ffa819c56f331fe8d9760c26abba27R48-R56).

Therefore in this case I'd recommend replacing the regex check with this in the guard chain:
`body("review._id").isAlphanumeric()`

Reply:
Noted, done 👍